console/src/api/auth.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import axios from "axios"
+import { apiClient } from "./client"
+import { navigate } from "@/lib/navigation"
+import { REALM_HEADER_NAME } from "@/lib/constants"
+import type { OAuthTokenResponse } from "@/types/api"
+
+// Always use relative URL to go through the proxy (dev server or production server)
+// This avoids CORS issues by proxying requests through the server
+// The server.ts proxy handles /api routes in production, and Vite handles them in development
+const TOKEN_URL = "/api/catalog/v1/oauth/tokens"
```

Comment:
Can we have a configurable token url for 3-rd party IDPs or different remote Polaris instance? Not a blocker. I'm fine with filing an issue and resolving it later.